kube/src/lib.rs

```diff
@@ -400,4 +397,80 @@ mod test {
         }
         Ok(())
     }
+
+    #[tokio::test]
+    #[ignore] // needs cluster (will create await a pod)
+    #[cfg(all(feature = "runtime"))]
+    async fn pod_can_await_conditions() -> Result<(), Box<dyn std::error::Error>> {
+        use crate::{
+            api::{DeleteParams, PostParams},
+            runtime::wait::{await_condition, conditions, delete::delete_and_finalize, Condition},
+            Api, Client,
+        };
+        use k8s_openapi::api::core::v1::Pod;
+        use std::time::Duration;
+        use tokio::time::timeout;
+
+        let client = Client::try_default().await?;
+        let pods: Api<Pod> = Api::default_namespaced(client);
+
+        // create busybox pod that's alive for at most 30s
+        let p: Pod = serde_json::from_value(serde_json::json!({
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": { "name": "busybox-kube4" },
+            "spec": {
+                "containers": [{
+                  "name": "busybox",
+                  "image": "busybox:1.34.1",
+                  "command": ["sh", "-c", "sleep 30"],
+                }],
+            }
+        }))?;
+
+        let pp = PostParams::default();
+        match pods.create(&pp, &p).await {
+            Ok(o) => assert_eq!(p.name(), o.name()),
+            Err(crate::Error::Api(ae)) => assert_eq!(ae.code, 409), // if we failed to clean-up
+            Err(e) => return Err(e.into()),                         // any other case if a failure
+        }
+
+        // Watch it phase for a few seconds
+        let is_running = await_condition(pods.clone(), "busybox-kube4", conditions::is_pod_running());
+        let _ = timeout(Duration::from_secs(15), is_running).await?;
+
+
+        // Verify we can get it
+        let pod = pods.get("busybox-kube4").await?;
+        assert_eq!(p.spec.as_ref().unwrap().containers[0].name, "busybox");
+
+        // Wait for a more complicated condition: ContainersReady AND Initialized
+        // TODO: remove these once we can write these functions generically
+        //fn is_containers_ready() -> impl Condition<Pod> {
+        //    |obj: Option<&Pod>| {
+        //        if let Some(o) = obj {
+        //            if let Some(s) = &o.status {
+        //                if let Some(conds) = &s.conditions {
+        //                    if let Some(pcond) = conds.iter().find(|c| c.type_ == "ContainersReady") {
+        //                        return pcond.status == "True";
+        //                    }
+        //                }
+        //            }
+        //        }
+        //        false
+        //    }
+        //}
+        //let is_fully_ready = conditions::is_pod_running().and(is_containers_ready());
+        //let _ = timeout(Duration::from_secs(15), is_fully_ready).await?;
```

Comment:
```suggestion
        //let _ = timeout(Duration::from_secs(15), await_condition(pods, "busybox-kube4", is_fully_ready)).await?;
```

The condition itself has no idea about what object you're querying for, or how to connect to the cluster.

Reply:
oh, no. :facepalm:
thank you.